smoderp2d/core/surface.py

```diff
@@ -301,40 +232,76 @@ def update_state1(ht_1, hcrit, state):
 
     :return: TODO
     """
-    if ht_1 > hcrit:
-        if state == 0:
-            return 1
+    state = ma.where(ht_1 > hcrit,ma.where(state == 0, 1, state), state)
+    # if ht_1 > hcrit:
+    #     if state == 0:
+    #         return 1
```

Comment:
Function `update_state1` is not used anywhere in the implicit solution. What is the purpose of modifying code that is not used anywhere here?